### PR TITLE
refactor: improved attributes-to-props, dom-to-react, utilities

### PR DESCRIPTION
**Pull request by ssi02014**
Release-As: 5.0.5

## What is the motivation for this pull request?

@remarkablemark 👋
I worked on improving `type`, and more intuitive code.

## What is the current behavior?

There is no change in behavior.

## What is the new behavior?

There is no change in behavior.

## Checklist:

- [x] [Conventional Commits](https://www.conventionalcommits.org/)
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: src/attributes-to-props.ts [hunk 1]
```diff
@@ -69,12 +69,12 @@ export default function attributesToProps(
...
           propName as UncontrolledComponentAttributes,
-        ) !== -1 &&
-        UNCONTROLLED_COMPONENT_NAMES.indexOf(
+        ) &&
+        UNCONTROLLED_COMPONENT_NAMES.includes(
           nodeName! as UncontrolledComponentNames,
-        ) !== -1 &&
+        ) &&
         !isInputValueOnly
       ) {
```
Comment (ssi02014) [inline src/attributes-to-props.ts:79]: I understood the code above to be checking whether a particular string is included or not, so I thought the `includes` method, which returns a `Boolean`, was more intuitive.

But obviously this is a matter of taste, so your opinion as a maintainer is even more important!

File: src/dom-to-react.ts [hunk 2]
```diff
@@ -95,7 +95,7 @@ export default function domToReact(
       props = attributesToProps(element.attribs, element.name);
     }
 
-    let children;
+    let children: ReturnType<typeof domToReact> | undefined;
```
Comment (ssi02014) [inline src/dom-to-react.ts:98]: I specified a type of `children`!

File: src/attributes-to-props.ts (hunk 1 above)
Comment (remarkablemark) [reply src/attributes-to-props.ts:79]: The reason `indexOf` was used before was to support older browsers. I think we should be safe to use `includes` now. What do you think? https://caniuse.com/?search=includes

File: src/dom-to-react.ts (hunk 2 above)
Comment (remarkablemark) [reply src/dom-to-react.ts:98]: Thanks

**Review by remarkablemark — APPROVED** [review]
Looks good, let me know what you think about https://github.com/remarkablemark/html-react-parser/pull/1148#discussion_r1380390016

**Comment by remarkablemark** [comment]
Release:

- https://github.com/remarkablemark/html-react-parser/pull/1149#issuecomment-1791073786

**Pull request merged**